### PR TITLE
Updated travis staged build following release meeting [Issue #109]

### PR DESCRIPTION
**Pull request by jusleg**
### WHAT kind of change does this PR introduce?
This pr allows our build to be segmented into multiple different stages which allows developers to know the stage at which the build has failed. Firstly, compilation of the program will take place once this stage is complete integration and ui tests will follow in their own discrete steps.

### HOW is this accomplished?
The `travis.yml` now triggers 5 discrete gradle tasks in sequential order which will allow developers to have a lot more insight on build failures. 

### Checklist
- [x] Docs have been added / updated to reflect the changes
- [x] I have reviewed and tested the changes :white_check_mark:
- [x] I have squashed my commits to have a reasonable amount of commits
- [x] All of the code I have written adhere to the coding conventions outlined in the coding conventions section of the wiki

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,12 +22,18 @@ before_script:
...
   - sed -i.bak '150,163d' $FLAKY_PATH
 
-env:
-  - TEST_SUITE=integration_tests
-  - TEST_SUITE=unit_tests
-
-script: ./testing $TEST_SUITE
+jobs:
+  include:
+    - stage: Compilev (Debug, UnitTest, AndroidTest & Release) 🌚
```
Comment (0matleb2) [inline .travis.yml]: Not sure if use of emoticons (🌚) follows the coding conventions.

Comment (jusleg) [reply .travis.yml]: It matches travis conventions: "Stages are identified by their names, which are composed of names and emojis." (Source: [docs.travis-ci.com](https://docs.travis-ci.com/user/build-stages/))

**Comment by almiche** [comment]
✅ Styling is adequate and follows coding conventions outlined by the [Kotlin Foundation](https://kotlinlang.org/docs/reference/coding-conventions.html)

✅ New methods and functionalities have appropriate tests (testing edge cases + good coverage)

✅ Branch has an acceptable amount of commits with descriptive names and

✅ PR description matches the code outlined a has mentions to issues

✅ This PR attempts to implement a specific feature (no code blob or epic pr)

Status: **Approved**

**Comment by jusleg** [comment]
@mehranhassani updated it following the discussion in the release meeting

File: .travis.yml (hunk 1 above)
Comment (jusleg) [reply .travis.yml]: @0matleb2 I had to do it 😉
